tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -577,11 +655,16 @@ private static DataType dataTypeOf(Object o) {
   }
 
   /**
-   * Returns the number of dimensions of a tensor of type dtype when represented by the object o.
+   * Return the number of dimensions of the tensor that object {@code o} represents as a tensor
+   * whose datatype is {@code dtype}. Normally this is the same as the number of dimensions of o
+   * itself, but is one smaller for tensors of strings.
+   *
+   * @param o The object to inspect. It must be a valid representation of the given data type.
+   * @param dtype The expected data type of the tensor.
    */
   private static int numDimensions(Object o, DataType dtype) {
     int ret = numArrayDimensions(o);
-    if (dtype == DataType.STRING && ret > 0) {
+    if (dtype == DataType.STRING) {
```

Comment:
Why remove the `&& ret > 0` from here?

Reply:
This was something I added in order to support initializing scalar UInt8 tensors from scalar byte values. But it's not clear we want to support that. Maybe should go into a separate PR?

Reply:
I guess a scalar byte value can still initialize a single-byte string, so I've restored the `ret > 0`